File: app/src/main/java/com/websarva/wings/android/zuboradiary/data/DayOfWeekStringConverter.java
package com.websarva.wings.android.zuboradiary.data;

import android.content.Context;

import androidx.annotation.NonNull;

import com.websarva.wings.android.zuboradiary.R;

import java.time.DayOfWeek;
import java.util.Objects;

/**
 *
 * Enum DayOfWeeK を用途に合わせた文字列に変換するクラス。
 * */
public class DayOfWeekStringConverter {
    private Context context;

    public DayOfWeekStringConverter(Context context) {
        this.context = context;
    }

    @NonNull
    public String toCalendarStartDayOfWeek(DayOfWeek dayOfWeek) {
        Objects.requireNonNull(dayOfWeek);

        int resId;
        switch (dayOfWeek) {
            case SUNDAY:
                resId = R.string.day_of_week_name_sunday;
                break;
            case MONDAY:
                resId = R.string.day_of_week_name_monday;
                break;
            case TUESDAY:
                resId = R.string.day_of_week_name_tuesday;
                break;
            case WEDNESDAY:
                resId = R.string.day_of_week_name_wednesday;
                break;
            case THURSDAY:
                resId = R.string.day_of_week_name_thursday;
                break;
            case FRIDAY:
                resId = R.string.day_of_week_name_friday;
                break;
            case SATURDAY:
                resId = R.string.day_of_week_name_saturday;
                break;
            default:
                return "";
        }
        return context.getString(resId);
    }

    @NonNull
    public String toDiaryListDayOfWeek(DayOfWeek dayOfWeek) {
        Objects.requireNonNull(dayOfWeek);

        int resId;
        switch (dayOfWeek) {
            case SUNDAY:
                resId = R.string.day_of_week_short_name_sunday;
                break;
            case MONDAY:
                resId = R.string.day_of_week_short_name_monday;
                break;
            case TUESDAY:
                resId = R.string.day_of_week_short_name_tuesday;
                break;
            case WEDNESDAY:
                resId = R.string.day_of_week_short_name_wednesday;
                break;
            case THURSDAY:
                resId = R.string.day_of_week_short_name_thursday;
                break;
            case FRIDAY:
                resId = R.string.day_of_week_short_name_friday;
                break;
            case SATURDAY:
                resId = R.string.day_of_week_short_name_saturday;
                break;
            default:
                return "";
        }
        return context.getString(resId);
    }
}

File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/utils/DayOfWeekStringConverter.kt
package com.websarva.wings.android.zuboradiary.ui.utils

import android.content.Context
import com.websarva.wings.android.zuboradiary.R
import java.time.DayOfWeek

/**
 * Enum java.time.DayOfWeeK を用途に合わせた文字列に変換するクラス。
 */
internal class DayOfWeekStringConverter(private val context: Context) {

    fun toCalendarStartDayOfWeek(dayOfWeek: DayOfWeek): String {
        val resId = when (dayOfWeek) {
            DayOfWeek.SUNDAY -> R.string.day_of_week_name_sunday
            DayOfWeek.MONDAY -> R.string.day_of_week_name_monday
            DayOfWeek.TUESDAY -> R.string.day_of_week_name_tuesday
            DayOfWeek.WEDNESDAY -> R.string.day_of_week_name_wednesday
            DayOfWeek.THURSDAY -> R.string.day_of_week_name_thursday
            DayOfWeek.FRIDAY -> R.string.day_of_week_name_friday
            DayOfWeek.SATURDAY -> R.string.day_of_week_name_saturday
            else -> throw IllegalArgumentException()
        }
        return context.getString(resId)
    }

    fun toDiaryListDayOfWeek(dayOfWeek: DayOfWeek): String {
        val resId = when (dayOfWeek) {
            DayOfWeek.SUNDAY -> R.string.day_of_week_short_name_sunday
            DayOfWeek.MONDAY -> R.string.day_of_week_short_name_monday
            DayOfWeek.TUESDAY -> R.string.day_of_week_short_name_tuesday
            DayOfWeek.WEDNESDAY -> R.string.day_of_week_short_name_wednesday
            DayOfWeek.THURSDAY -> R.string.day_of_week_short_name_thursday
            DayOfWeek.FRIDAY -> R.string.day_of_week_short_name_friday
            DayOfWeek.SATURDAY -> R.string.day_of_week_short_name_saturday
            else -> throw IllegalArgumentException()
        }
        return context.getString(resId)
    }
}
